docs/content/utilities/batch.mdx

```diff
@@ -0,0 +1,95 @@
+---
+title: Batch
+description: Utility
+---
+
+import Note from "../../src/components/Note"
+
+The batch utility provides an abstraction to process a batch event. Useful for lambda integrations with [AWS SQS](https://aws.amazon.com/sqs/), [AWS Kinesis](https://aws.amazon.com/kinesis/) and [AWS DynamoDB Streams](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Streams.html).
```

Comment:
Got your point, let me try to explain mine. As you said in one of the comments, the selling point is definitely "partial batch processing" but *what if* we also provided some a default batch processor (without the "partial behavior") ? We would be able to offer an interface for anyone doing batch processing. Personally I'm always doing code similar to this:
```python
def handle_record(record):
    return record

def handler(event, context):
   records = event["Records"]
   for record in records:
      do_something()
   return True
```
*If* we follow the suggest approach, we'll be able to provide a standard not only from sqs, but all current/future providers.
```python
from aws_lambda_powertools.utilities.batch import BatchProcessor
# BatchProcessor doesn't exist, we may be able to create now or in a next PR

def handle_record(record)
   return record

@batch_processor(handle_record, BatchProcessor())
def handler(event, context):
    return True
```
Do you think is a good idea ? Not for now, of course :).

Reply:
I like the idea, a sample that would benefit from this would be an AWS Lambda function that handles SNS for example. But as long as that functionality is not in the codebase I have to agree with @heitorlessa, currently we only do partial SQS processing so the documentation should reflect that.